### ISSUE TITLE
boot.tmpOnTmpfs fails to mount

### ISSUE THREAD

**Issue by sagehane** [NixOS/nixpkgs#109118]
**Describe the bug**
I recently updated to the latest `nixos-unstable` channel and attempted to rebuild my system, and came across this message:
```
setting up tmpfiles
warning: the following units failed: tmp.mount

● tmp.mount - /tmp
     Loaded: loaded (/nix/store/d5ylr9h5n2wyzalsg7v5vgqd3dnzh6gy-unit-tmp.mount/tmp.mount; linked; vendor preset: enabled)
     Active: failed (Result: exit-code) since Tue 2021-01-12 23:03:11 JST; 17ms ago
      Where: /tmp
       What: tmpfs
         IP: 0B in, 0B out
        CPU: 777us

Jan 12 23:03:11 nixos systemd[1]: Mounting /tmp...
Jan 12 23:03:11 nixos mount[9780]: mount: /tmp: special device tmpfs does not exist.
Jan 12 23:03:11 nixos systemd[1]: tmp.mount: Mount process exited, code=exited, status=32/n/a
Jan 12 23:03:11 nixos systemd[1]: tmp.mount: Failed with result 'exit-code'.
Jan 12 23:03:11 nixos systemd[1]: Failed to mount /tmp.
```

**To Reproduce**
1. Set `boot.tmpOnTmpfs` to true
2. Run `nixos-rebuild test`

**Expected behavior**
The rebuild shouldn't fail, as was the case before I updated.

**Additional context**
I use XFS for my root partition, if that's relevant.

**Notify maintainers**

I had no idea what to put here.

**Metadata**
```
$ nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-linux"`
 - host os: `Linux 5.10.6, NixOS, 21.03pre262347.257cbbcd3ab (Okapi)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.3.10`
 - channels(root): `"nixos-21.03pre262347.257cbbcd3ab"`
 - channels(plumeus): `""`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`
 ```
Maintainer information:
```yaml
# a list of nixos modules affected by the problem
module:
boot.tmpOnTmpfs
```

**Comment by jshholland** [comment]
See PRs #108860 and #108881

**Comment by iblech** [comment]
TL;DR Until a proper fix, add this to your configuration.nix:

    systemd.additionalUpstreamSystemUnits = "tmp.mount";

**Comment by sagehane** [comment]
Thanks very much for the info. I forgot to check PRs and probably wouldn't have figured out which one to look out for otherwise.
Just a minor correction regarding the systemd option, but seems like it should be:
```
systemd.additionalUpstreamSystemUnits = [ "tmp.mount" ];
```
I guess I'll close this issue and follow the PRs, thanks.

**Comment by poscat0x04** [comment]
This issue still exist and can potentially break one's system

**Comment by poscat0x04** [comment]
might want to pin this as well because of its severity

**Comment by alyssais** [comment]
Should have been fixed by #108860 AIUI.
